server/testkit/tests/identity_verification_tests.rs

```diff
@@ -0,0 +1,448 @@
+static INVALID_USER_NAME: &'static str = "invalid_user";
+
+static VALID_USER_A_UUID: &'static str = "d2b496bd-8493-47b7-8142-f568b5cf47e1";
+static VALID_USER_A_NAME: &'static str = "valid_user_A";
+
+static VALID_USER_B_UUID: &'static str = "d2b496bd-8493-47b7-8142-f568b5cf47e2";
+static VALID_USER_B_NAME: &'static str = "valid_user_B";
+
+fn get_identity_verification_endpoint(other_id: &str) -> String {
+    format!("/v1/person/{other_id}/_identify_user")
+}
+
+fn get_url(addr: &str, id: &str) -> String {
+    format!("{addr}/{}", get_identity_verification_endpoint(id))
+}
+
+// UNAUTHENTICATED CASES:
+#[kanidmd_testkit::test]
+async fn test_not_authenticated(rsclient: KanidmClient) {
+    // basically here we try a bit of all the possible combinations while unauthenticated to check it's not working
+    setup_server(&rsclient).await;
+    create_valid_user_a(&rsclient).await;
+    rsclient.logout();
+    let res: Result<IdentifyUserResponse, Error> = make_request(
+        &rsclient,
+        IdentifyUserRequest::Start,
+        Some(VALID_USER_A_NAME.to_string()),
+    )
+    .await;
+    assert!(res.is_err());
+    if let Err(err) = res {
+        assert_eq!(err.status().unwrap(), StatusCode::FORBIDDEN)
+    }
+    let res: Result<IdentifyUserResponse, Error> = make_request(
+        &rsclient,
+        IdentifyUserRequest::DisplayCode,
+        Some(VALID_USER_A_NAME.to_string()),
+    )
+    .await;
+    assert!(res.is_err());
+    if let Err(err) = res {
+        assert_eq!(err.status().unwrap(), StatusCode::FORBIDDEN)
+    }
+    let res: Result<IdentifyUserResponse, Error> = make_request(
+        &rsclient,
+        IdentifyUserRequest::SubmitCode { other_totp: 123456 },
+        None,
+    )
+    .await;
+    assert!(res.is_err());
+    if let Err(err) = res {
+        assert_eq!(err.status().unwrap(), StatusCode::FORBIDDEN)
+    }
+}
+
+// TEST ON ERROR OUTCOMES
+// These tests check that no matter what we request if the identity verification feature
+// has an issue the correct error will be returned!
+// Each test is named after the error it should return
+#[kanidmd_testkit::test]
+async fn test_identity_verification_unavailable(rsclient: KanidmClient) {
+    setup_server(&rsclient).await;
+    create_invalid_user(&rsclient).await;
+    login_with_user(&rsclient, INVALID_USER_NAME).await;
+    let response: Result<IdentifyUserResponse, Error> =
+        make_request(&rsclient, IdentifyUserRequest::Start, None).await;
+    assert!(response.is_ok());
+    assert_eq!(
+        response.unwrap(),
+        IdentifyUserResponse::IdentityVerificationUnavailable
+    );
+    let res: Result<IdentifyUserResponse, Error> = make_request(
+        &rsclient,
+        IdentifyUserRequest::Start,
+        Some(VALID_USER_A_NAME.to_string()),
+    )
+    .await;
+    assert!(res.is_ok());
+    assert_eq!(
+        res.unwrap(),
+        IdentifyUserResponse::IdentityVerificationUnavailable
+    );
```

Comment:
space :) lets the codes breaaaathe. 

It helps the reader deliniate one case from the next else it's hard to know if it's all part of one blob or separate.